### PR TITLE
Reduce PHP compatibility from 5.6 to 5.5.9

### PR DESCRIPTION
**Pull request by johnkary**
> NOTE: This is a work-in-progress PR to allow Travis.CI running and visibility to code. Any and all commits will be squashed/rebased at will.

Reduce PHP compatibility to 5.5.9 to accommodate using Haystack in Drupal 8. PHP 5.5.9 is also the version shipped with Ubuntu 14.04 LTS.

Fixes Issue #10.
#### Changes
- Reduce `phpunit/phpunit` to 4.8 instead of 5.0 because 5.0 required PHP 5.6+.

### CODE REVIEW

**Comment by johnkary** [comment]
To get the right dependencies I had to run `composer update` when using PHP 5.5.9. To keep PHP 5.5 compatibility will we have to update dependencies only from systems with PHP 5.5? I ran `composer update phpunit/phpunit` after downgrading to PHPUnit 4.8 but it pulls in PHPUnit dependencies that require PHP >= 5.6. So Travis was failing for PHP 5.5 builds.

I don't have PHP 5.5 locally so I created a server on a remote hosting provider, installed 5.5, ran composer update to get the right dependencies. How do other projects keep dependencies targeted at the right versions when their dev machine has a higher version? VMs?

File: src/Functional/HArrayFilterWithKey.php [hunk 1]
```diff
@@ -20,6 +20,16 @@ public function __construct(array $arr)
...
+        if (!defined('ARRAY_FILTER_USE_KEY')) {
+            $return = [];
+            foreach ($this->arr as $k => $v) {
+                if (call_user_func($func, $k)) {
+                    $return[$k] = $v;
+                }
+            }
+            return $return;
+        }
+
```
Comment (johnkary) [inline src/Functional/HArrayFilterWithKey.php:32]: Not totally thrilled with the polyfill implementations but tests pass! If there are other ways to implement this I'm open.

**Comment by johnkary** [comment]
Travis tests pass! Could you review so we could merge this before my Drupal talk? 👻 

I'm in favor of squashing all of these commits into 1 commit once this is approved. Otherwise intermediate commits would either install wrong dependencies or the test suite would fail.

**Comment by ericpoe** [comment]
To pull in the proper dependencies when the dev box has a later version of PHP, we should add the following to `composer.json`

``` php
"config": {
  "platform": {
    "php": "5.5.9"
  }
},
```

**Comment by ericpoe** [comment]
Very good!

Please also update the minimum requirement in [docs/manual.md](../blob/master/docs/manual.md)

Once you get the `config` section added to `composer.json`, please squash the commits. I'll then merge.

**Comment by johnkary** [comment]
@ericpoe Changes complete. Anything else to make this ready?

**Comment by ericpoe** [comment]
@johnkary Looks great! Thanks!

**Pull request merged**